core/src/impl/Kokkos_Stacktrace.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef KOKKOS_STACKTRACE_HPP
+#define KOKKOS_STACKTRACE_HPP
+
+#include <functional>
+#include <ostream>
+#include <string>
+
+namespace Kokkos {
+namespace Impl {
+
+/// \brief Return the demangled version of the input symbol, or the
+///   original input if demangling is not possible.
+std::string demangle(const std::string& name);
+
+/// \brief Save the current stacktrace.
+///
+/// You may only save one stacktrace at a time.  If you call this
+/// twice, the second call will overwrite the result of the first
+/// call.
+void save_stacktrace();
```

Comment:
As above, I'd rather this just returned a `Stacktrace`. There's no real reason to use a global variable here when the user can do that themselves just as easily.

Reply:
@dhollman I'm not sure if that was the intent, but if you want to capture or access a stack trace from a signal handler, you'll need some kind of global variable with static storage.

Reply:
We leave this for now, since this might be a valid use-case. Also this is all inside "Impl" namespace for now, so we can change it later.

Reply:
sure. We'll leave it for now, I guess